### PR TITLE
Install traffic-mesh via npm package

### PR DESCRIPTION
**Pull request by NeoLegends**
**- Summary**

This configures the CLI to load `traffic-mesh` via npm instead of via downloading it manually the first time `netlify dev` is run.

~~The package itself is quite heavy right now (40mb) because it contains the binaries for all operating systems, but we can always improve on that later by shipping a stripped package that downloads the right binaries on installation. For the MVP I wanted to avoid the additional moving parts this brings.~~

**- Test plan**

This should not change any behavior, so the existing tests should continue working.

**- Description for the changelog**

Install `traffic-mesh` component via npm package.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by ehmicky** [comment]
> The package itself is quite heavy right now (40mb) because it contains the binaries for all operating systems

Can anything be done to remove some bytes, such as removing debugging symbols, etc.?

> but we can always improve on that later by shipping a stripped package that downloads the right binaries on installation.

The main concern with a bigger binary size is that this would add time when installing `netlify-cli`. Downloading on `postinstall` would add even more time, so I think the current solution might be best.

**Comment by NeoLegends** [comment]
> Can anything be done to remove some bytes, such as removing debugging symbols, etc.?

Yeah the binaries aren‘t stripped yet.

**Comment by NeoLegends** [comment]
Follow-up: Stripping binaries removes any useful information from their stack traces (in case of panics), so I'd opt to leave the debug symbols in them for now and only remove those later.

**Comment by NeoLegends** [comment]
Follow up 2: I'm working on leveraging optional dependencies to make npm only download the package you need for your platform.

This gives us the benefit of using npm for all the I/O and has only marginally increased complexity.

See https://github.com/netlify/traffic-mesh/pull/1217

**Comment by NeoLegends** [comment]
Download size per platform is now:

- macOS: 12.8MB
- Linux: 16.6MB
- Windows: 11.2MB

**Comment by NeoLegends** [comment]
Hey @ehmicky @erezrokah, just wanted to ask your status on this PR. I'm happy to move forward with this PR and #1461, but I'm not sure about your schedule and / or whether you're currently blocked on other things. :)

I think I'll have the capacity to finish the integration next week, so up to you, whenever you have time.

**Review by erezrokah — APPROVED** [review]
This looks good to me, but let's wait on https://github.com/netlify/cli/pull/1505 before merging.

**Comment by erezrokah** [comment]
`npm ci` failed:
https://github.com/netlify/cli/runs/1341703190?check_suite_focus=true#step:4:4
Going to re-run it.

**Pull request merged**
